commit a6876b6a9252e8afe0e2b344b7713fe8da739059
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2019-09-27T08:05:13-07:00 (10 days ago)
    
    joyent/sdc-cloudapi#34 Instance resize fails with differing traits error
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/docs/index.md b/docs/index.md
index dde36cc9..08c9e418 100644
--- a/docs/index.md
+++ b/docs/index.md
@@ -841,6 +841,12 @@ Note that a `Triton-Datacenter-Name` response header was added in 9.2.0.
 
 The section describes API changes in CloudAPI versions.
 
+## 9.8.2
+- Fixed instance resize from failing during package trait check.
+
+## 9.8.1
+- Added migration finalize API [Migrations](#migrations).
+
 ## 9.8.0
 - Added support for retrieving account [provisioning limits](#GetAccountLimits).
 
diff --git a/lib/machines.js b/lib/machines.js
index ae64e3c1..3acba6b5 100644
--- a/lib/machines.js
+++ b/lib/machines.js
@@ -238,6 +238,9 @@ function translate(machine, req) {
         var packages = req.packages.filter(function (d) {
             return (d.uuid === machine.billing_id);
         });
+        if (req.params.action === 'resize') {
+            req.vmPkg = packages[0];
+        }
         msg.package = packages[0] ? packages[0].name : '';
 
         if (machine.brand === 'bhyve') {
@@ -2146,10 +2149,10 @@ function resize(req, res, next) {
                     pkgRam, imgReq.max_ram));
     }
 
-
-    if (!jsprim.deepEqual(req.pkg.traits, req.machine.package.traits)) {
+    // Note that req.vmPkg was set during the machine translate() call.
+    if (req.vmPkg && !jsprim.deepEqual(req.pkg.traits, req.vmPkg.traits)) {
         return next(new InvalidArgumentError(
-            'Resizing to package with differing traits not supported'));
+            'Resizing to a package with differing traits not supported'));
     }
 
     var callback = updateCallback(req, res, next),
diff --git a/package.json b/package.json
index 7fd7dd8a..872ec45d 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "cloudapi",
     "description": "Triton CloudAPI",
-    "version": "9.8.1",
+    "version": "9.8.2",
     "author": "Joyent (joyent.com)",
     "private": true,
     "engines": {
